### ISSUE TITLE
performance degrade after upgrading plugin 

### ISSUE THREAD

**Issue by lukavucko** [jfrog/build-info#230]
upgrading from 4.4.9 to to any higher version causes serious performance degrade. For publishing 600 projects it lasts 10 minutes longer than usual. Also this task is org.jfrog.gradle.plugin.artifactory.task.DeployTask is executing, which was not a case before.

Build scan examples:
![Version  4 4 9](https://user-images.githubusercontent.com/34003679/58002703-5f7f8880-7adf-11e9-919f-6c4c39ae1197.png)

![Version 4 9 6](https://user-images.githubusercontent.com/34003679/58002740-7de58400-7adf-11e9-88d2-e7994bb30ed7.png)

**Comment by eyalbe4** [comment]
@lukavucko,

Can you see which task takes longer now? 
How can we see this performance degradation?

As for the *DeployTask*, we added it in order to resolve a concurrency issue. Before the *DeployTask*, one of the *artifactoryTask* tasks used be elected, to perform the deployment. The approach was to elect the last tasl to run. However, in parallel mode, this election process was not consistent anymorte and caused issues. The newly added *DeployTask* completely resolved this issue.

**Comment by lukavucko** [comment]
You can see on two pictures above.
On first picture you can see that task artifactoryDeploy takes 24 min to execute.
On second picture you can see that task artifactoryPublish takes 14 min to execute.
With new updating of artifactoryPlugin to 4.9.6 I have new task artifactoryDeploy which takes 24 min and it is 10 minutes longer than artifactoryPublish task which is in older version of artifactoryPlugin.

**Comment by eyalbe4** [comment]
@lukavucko,
Assuming that the results you see are consistent, the only thing that comes to mind, which can explain this change, is that perhaps you were using the parallel execution of gradle. This causes some of the tasks to run in parallel, which lead to faster deployment.
Now that all the deployment is performed by one task, which depends on all the other tasks to finish their execution, the deployment is sequential.
Could this be the cause what you're seeing?

**Comment by lukavucko** [comment]
Maybe it is the case what you said, thank you for your answer.
I will investigate that.

**Comment by lukavucko** [comment]
After some tests we found out the performance problem is not related with the versions.
The problem is with Gradle 5.4 we are not able anymore to get rid of builtOn dependencies on publications artifacts. The reason for getting rid of these dependencies is performance issue.
You can see here how we cut the builtOn dependencies:
if (rootProject.hasProperty('disableBuiltByOnPublicationArtifacts')) {
        gradle.projectsEvaluated {
            rootProject.allprojects { p ->
                p.pluginManager.withPlugin('maven-publish') {
                    p.publishing.publications.each { pub ->
                        logger.debug("-- Disable builtBy on artifacts of publication '${pub.name}'.")
                        pub.artifacts.each { artifact ->
                            **artifact.buildDependencies.setValues**([]) -- in Gradle 5.4 the model is read only
                        }
                    }
                }
            }
        }
    }
On these two pictures you can see the results:

![with_cutting_dependencies](https://user-images.githubusercontent.com/34003679/58178525-a8ce0480-7ca6-11e9-8da4-308af84ad996.png)

![without_cutting](https://user-images.githubusercontent.com/34003679/58178844-3a3d7680-7ca7-11e9-9157-006d41820395.png)

Build scan pictures show the differences between cutting and without cutting: 
![Build scan_with_cutting](https://user-images.githubusercontent.com/34003679/58179274-28100800-7ca8-11e9-8594-3bea3976919c.png)

![Build scan_without_cutting](https://user-images.githubusercontent.com/34003679/58179294-30684300-7ca8-11e9-8719-44ee92b21fc4.png)

We have exactly the same scenario after the upgrade on Gradle 5.4 and artifactoryPluginVersion 4.9.6.

Is it possible to configure publications artifacts without builtOn dependencies in some other way? Because in this line: **artifact.buildDependencies.setValues([])**, the model in Gradle 5.4 is read only.
